cdap-app-templates/cdap-etl/cdap-etl-api/src/main/java/co/cask/cdap/templates/etl/api/batch/AbstractBatchSink.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.templates.etl.api.batch;
+
+import co.cask.cdap.templates.etl.api.StageConfigurer;
+import co.cask.cdap.templates.etl.api.StageLifecycle;
+
+import java.io.IOException;
+
+/**
+ * Batch Sink forms the last stage of a Batch ETL Pipeline.
+ * @param <I> Object sink operates on
+ * @param <KEY> Batch Output Key class
+ * @param <VALUE> Batch Output Value class
+ */
+public abstract class AbstractBatchSink<I, KEY, VALUE> implements StageLifecycle {
+
+  private BatchContext context;
+
+  /**
+   * Configure the Sink.
+   * @param configurer {@link StageConfigurer}
+   */
+  public void configure(StageConfigurer configurer) {
+    configurer.setName(this.getClass().getSimpleName());
+  }
+
+  /**
+   * Prepare the Batch Job.
```

Comment:
the difference between prepareJob and initialize is not clear to me.  I assume this gets called when we're preparing the mapreduce job? What would you do in initialize that would not be done here?